core/src/main/java/io/questdb/std/str/DirectByteCharSink.java

```diff
@@ -33,71 +33,91 @@
 
 import java.io.Closeable;
 
-public class DirectByteCharSink extends AbstractCharSink implements Mutable, ByteSequence, Closeable {
-    private final long initialCapacity;
-    private long capacity;
-    private long hi;
-    private long lo;
-    private long ptr;
+public class DirectByteCharSink extends AbstractCharSink implements Mutable, ByteSequence, Closeable, BorrowableDirectByteSink {
```

Comment:
I'd better introduce a separate class instead of building on top of `DirectByteCharSink`. The reasons are the following. First, I'm going to remove this class in favor of `DirectUtf8Sink` soon. Second, we use this class in many places in ILP and other interfaces and we don't share the sink memory with native code there, so asking all users to pay for the extra JNI call when the sink grows may be too much.

Reply:
Let's keep your changes as is. I'll port your changes to the new class. As for the JNI overhead, once the sink grows large enough, there would be no overhead.